src/HDF5Msg.chpl

```diff
@@ -24,1759 +26,1622 @@ module HDF5Msg {
-            return new MsgTuple(errorMsg, MsgType.ERROR);
-        }
+    config const SINGLE_FILE: int = 0;
+    config const MULTI_FILE: int = 1;
 
-        // If the filename represents a glob pattern, retrieve the locale 0 filename
-        if isGlobPattern(filename) {
-            // Attempt to interpret filename as a glob expression and ls the first result
-            var tmp = glob(filename);
-            h5Logger.debug(getModuleName(),getRoutineName(),getLineNumber(),
-                      "glob-expanded filename: %s to size: %i files".format(filename, tmp.size));
+    require "c_helpers/help_h5ls.h", "c_helpers/help_h5ls.c";
+    private extern proc c_get_HDF5_obj_type(loc_id:C_HDF5.hid_t, name:c_string, obj_type:c_ptr(C_HDF5.H5O_type_t)):C_HDF5.herr_t;
+    private extern proc c_strlen(s:c_ptr(c_char)):c_size_t;
+    private extern proc c_incrementCounter(data:c_void_ptr);
+    private extern proc c_append_HDF5_fieldname(data:c_void_ptr, name:c_string);
 
-            if tmp.size <= 0 {
-                var errorMsg = "Cannot retrieve filename from glob expression %s, check file name or format".format(filename);
-                h5Logger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
-                return new MsgTuple(errorMsg, MsgType.ERROR);
-            }
+    /*
+     * Returns the HDF5 data type corresponding to the dataset, which delegates
+     * to getHDF5Type for all datatypes supported by Chapel. For datatypes that
+     * are not supported by Chapel, getDataType encapsulates logic to retrieve
+     * the HDF5 data type.
+     */
+    proc getDataType(A) : C_HDF5.hid_t {
+        var dType : C_HDF5.hid_t;
             
-            // Set filename to globbed filename corresponding to locale 0
-            filename = tmp[tmp.domain.first];
+        if A.eltType == bool {
+            return C_HDF5.H5T_NATIVE_HBOOL;
+        } else {
+            return getHDF5Type(A.eltType);
         }
-        
-        // Check to see if the file exists. If not, return an error message
-        if !exists(filename) {
-            var errorMsg = "File %s does not exist in a location accessible to Arkouda".format(filename);
-            h5Logger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
-            return new MsgTuple(errorMsg,MsgType.ERROR);
-        } 
+    }
 
-        if !isHdf5File(filename) {
-            var errorMsg = "File %s is not an HDF5 file".format(filename);
-            h5Logger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
-            return new MsgTuple(errorMsg,MsgType.ERROR);
+    /*
+        Returns the C_HDF5.hid_t corresponding to the provided Chapel type
+    */
+    proc getDataType(type t) : C_HDF5.hid_t {
+        if t == bool {
+            return C_HDF5.H5T_NATIVE_HBOOL;
         }
-        
-        try {
+        else {
+            return getHDF5Type(t);
+        }
+    }
```

Comment:
let's double check that we're actually using this helper

Reply:
This is being used on ln 390. This usage in `writeLocalDset` mimic the call to `getDataType` from ln 65 that takes in an array when writing to a distributed file set. We could add it to a few spots where we `var dtype = C_HDF5.H5T_NATIVE_HBOOL` but that seems like overkill in situations where we know that value is to be set.